multicluster/apis/multicluster/v1alpha1/multiclusterconfig_types.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	config "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+//+kubebuilder:object:root=true
+
+// MultiClusterConfig is the Schema for the multiclusterconfigs API
+type MultiClusterConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// ControllerManagerConfigurationSpec returns the contfigurations for controllers
+	config.ControllerManagerConfigurationSpec `json:",inline"`
```

Comment:
Probably I do not understand the new configuration pattern. What config parameters in ControllerManagerConfigurationSpec apply to our case? Users need to create a MultiClusterConfig CR to configure multicluster-controller?

Reply:
we will provide auto-generated yaml file to customer based on different roles, so there is no need for user to create it. due to @aravindakidambi have the latest manifests and auto-generated scripts in her PR, so I didn't add those old one here.

Reply:
it eventually will be a ConfigMap like here https://github.com/abhiraut/antrea/blob/staging/multi-cluster/multicluster/build/yamls/antrea-multicluster-member-only.yml#L1633,  so the usage it's just like antrea-agent.conf. we won't create a CRD for it, so it's not allowable to create them via CR directly.

Reply:
So the CRD is for kubebuilder to generate the ConfigMap? Does MC controller uses all parameters in ControllerManagerConfigurationSpec?